web/src/app/components/Products/Products.tsx

```diff
@@ -29,15 +29,45 @@ const currencyMap: { [k in Currency]: string } = {
   Pound: 'GBP'
 }
 
+const columnsToSortByMap: { [k in keyof Book]: ColumnType } = {
+  exchange: 'Exchange',
+  product: 'Product',
+  bid: 'Bid',
+  ask: 'Ask',
+  value: 'Value'
+}
+
+const sortByToColumnsMap: { [k in ColumnType]: keyof Book } = {
+  Exchange: 'exchange',
+  Product: 'product',
+  Bid: 'bid',
+  Ask: 'ask',
+  Value: 'value'
+}
+
+type SortBy = ColumnType;
+type SortDir  = 'asc'| 'desc';
```

Comment:
Questo `SortDir` potresti definirlo come `type` dentro a un file `model/models.ts` (dove di solito definiamo tutto quello che ci serve in più rispetto alle cose generate da metarpheus) e importarlo direttamente da `'model'`. Anche se in questo caso, con una sola tabella, non darebbe benefici dato che lo usi solo qua.

Reply:
Sì in questo caso lo lascerei come eventuale refactoring da farsi eventualmente in futuro quando ce ne fosse bisogno, ora come ora non sono previste altre tabelle (e quindi non credo ce ne saranno mai).
Per quanto riguarda i ` t.Any ` sono dovuti al fatto che ` metarpheus ` ora usa ` io-ts ` mentre ` tcomb ` è ancora usato in molti file, quindi i ` t.Any ` sono dei workaround